go/vt/vtctl/reparentutil/durability_funcs.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reparentutil
+
+import (
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+	"vitess.io/vitess/go/vt/vtctl/reparentutil/promotionrule"
+)
+
+// SemiSyncAckersForPrimary returns the list of tablets which are capable of sending Semi-Sync Acks for the given primary tablet
+func SemiSyncAckersForPrimary(primary *topodatapb.Tablet, allTablets []*topodatapb.Tablet) (semiSyncAckers []*topodatapb.Tablet) {
+	for _, tablet := range allTablets {
+		if topoproto.TabletAliasEqual(primary.Alias, tablet.Alias) {
+			continue
+		}
+		if IsReplicaSemiSync(primary, tablet) {
+			semiSyncAckers = append(semiSyncAckers, tablet)
+		}
+	}
+	return
+}
+
+// RevokeForTablet checks whether we have reached enough tablets such that the given primary eligible tablet cannot accept any new writes
+func RevokeForTablet(primaryEligible *topodatapb.Tablet, tabletsReached []*topodatapb.Tablet, allTablets []*topodatapb.Tablet) bool {
+	// if we have reached the primaryEligible tablet and stopped its replication, then it will not
+	// accept any new writes
```

Comment:
Is this correct? Just because replication is stopped does not mean it can't accept writes, that is only true when semi-sync is enabled.

Reply:
For all tablets in this list that think they are Primary type, in `StopReplicationAndBuildStatusMaps`, we call `demotePrimary` on them, which succeeds only after setting the tablet to rdonly. So all the tablets in this list would not have just stopped replication but would also be read only, so just checking that the primary tablet is in the list is sufficient

Reply:
Hmm.. so usage of this function is dependent on some pre-condition being satisfied.
Is there a reason it has to be a separate function in a separate file? If it is being used only once, it can as well be just a code block, and then you don't have to worry about whether callers are satisfying the pre-condition.
If it _MUST_ be in a separate file for some good reason, you should make these two `Revoke` functions unexpected, so that they cannot be used outside the package. `s/Revoke/revoke`.

Reply:
Yes that's a good point. The only reason that I have kept them as separate function is for testing and readability. It makes it much easier to have unit tests for these individual functions along with unit tests for the larger StopReplicationAndBuildStatusMap since that requires a larger setup with a tablet manager etc. 
Just to be extra careful, I'll add a comment as well there stating the assumption that the function has and unexport it

Reply:
Fixed